test/report.js

```diff
@@ -0,0 +1,140 @@
+                });
+            let token = res.body.token;
+            res = await chai.request(server)
+                .post("/api/reports")
+                .send(requestBody)
+                .set('Authorization', token);
+            res.should.have.status(201);
+        });
+
+        it('it should not allow creation of a report with missing required data', async () => {
+            let requestBody = {}
+            let res = await chai.request(server)
+                .post("/api/authenticate/login")
+                .send({
+                    username: patientOne.username,
+                    password: patientOne.password
+                });
+            let token = res.body.token;
+            res = await chai.request(server)
+                .post("/api/reports")
+                .send(requestBody)
+                .set('Authorization', token);
+            res.should.have.status(400);
+        });
+    });
+
+    describe("/GET reports", () => {
+        beforeEach(async () => {
+            await userService.signUp(patientOne.username, patientOne.password, patientOne.lastName, patientOne.firstName, patientOne.email, patientOne.type);
+            await userService.signUp(patientTwo.username, patientTwo.password, patientTwo.lastName, patientTwo.firstName, patientTwo.email, patientTwo.type);
+        
+            let users = await User.findAll({
+                where: {},
+                attributes: ['id', 'username', 'password'],
+                order: [['id', 'DESC']]
+            });
+            let userOne = users[0];
+            let userTwo = users[1];
+
+            await reportService.createUserReport(userOne.id,userTwo.id);
+        });
+
+        it('it should retrieve all reports', async () => {
+            let res = await chai.request(server)
+                .post("/api/authenticate/login")
+                .send({
+                    username: patientOne.username,
+                    password: patientOne.password
+                });
+            let token = res.body.token;
+
+            res = await chai.request(server)
+                .get("/api/reports")
+                .set("Authorization", token);
+            
+            let body = res.body;
+            assert.exists(body);
+            assert.equal(1,body.length);
+        });
+
+        it('it should retrieve all pending reports', async () => {
+            let res = await chai.request(server)
+                .post("/api/authenticate/login")
+                .send({
+                    username: patientOne.username,
+                    password: patientOne.password
+                });
+            let token = res.body.token;
+
+            res = await chai.request(server)
+                .get("/api/reports?status=pending")
+                .set("Authorization", token);
+            
+            let body = res.body;
+            assert.exists(body);
+            assert.equal(1,body.length);

```

Comment:
this works, but should we do `body.should.have.length(1)` to stay consistent with other tests?

Reply:
I also changed the attributes test to use body.should.have.length